python/ray/tests/test_actor_failures.py

```diff
@@ -29,6 +30,32 @@ def ray_init_with_task_retry_delay():
     ray.shutdown()
 
 
+@pytest.fixture
+def temp_files():
+    """Create temporary files and ensure cleanup."""
```

Comment:
is the purpose of making this a factory so that you can use it multiple times in the same test function?

if so let's name it appropriately and add type hint
```suggestion
@pytest.fixture
def tempfile_factory() -> Generator[Callable[[], str]]:
    """Yields a factory function to generate tempfiles that will be deleted after the test run."""
```